admin/admin-links.php

```diff
@@ -272,4 +276,47 @@ public function get_data_from_new_post_translation_request( string $post_type ):
 			'new_lang'  => $lang,
 		);
 	}
+
+	/**
+	 * Returns some data (`from_post` and `new_lang`) from the current request.
+	 *
+	 * @since 3.7
+	 *
+	 * @return array {
+	 *     @type WP_Post      $from_post The source post.
+	 *     @type PLL_Language $new_lang  The target language.
+	 * }
+	 *
+	 * @phpstan-return array{}|array{from_post: WP_Post, new_lang: PLL_Language}|never
+	 */
+	public function get_data_from_new_media_translation_request() {
+		if ( ! $this->options['media_support'] ) {
+			return array();
+		}
+
+		if ( ! isset( $_GET['action'], $_GET['_wpnonce'], $_GET['from_media'], $_GET['new_lang'] ) || 'translate_media' !== $_GET['action'] ) {
+			return array();
+		}
+
+		check_admin_referer( 'translate_media' );
+
+		$post_id   = (int) $_GET['from_media'];
+		$lang_slug = sanitize_key( $_GET['new_lang'] );
+
+		if ( $post_id <= 0 || empty( $lang_slug ) ) {
+			return array();
+		}
+
+		$post = get_post( $post_id );
+		$lang = $this->model->get_language( $lang_slug );
+
+		if ( empty( $post ) || empty( $lang ) ) {
+			return array();
+		}
+
+		return array(
+			'from_post' => $post,
+			'new_lang'  => $lang,
+		);
```

Comment:
Maybe we could share this part with `get_data_from_new_post_translation_request()`.

Ex:
```php
	private function get_translation_request_data_objects( int $post_id, string $lang_slug ): array {
		if ( $post_id <= 0 || empty( $lang_slug ) ) {
			return array();
		}

		$post = get_post( $post_id );
		$lang = $this->model->get_language( $lang_slug );

		if ( empty( $post ) || empty( $lang ) ) {
			return array();
		}

		return array(
			'from_post' => $post,
			'new_lang'  => $lang,
		);
	}
```

Reply:
The returned key should be `from_media` instead of `from_post`, am I right?

Reply:
STOP RUINING EVERYTHING! 😭

Reply:
On second thought, we could keep `from_post` for both, it would require only few adjustments for the media 🤔

Reply:
After working on https://github.com/polylang/polylang-pro/pull/2274 which depends on this very PR, I think we should keep it that way.
After all a media is a post, and it allows to reuse completely what's done in `PLL_ACF_Post`.

Reply:
I intentionally kept `from_post` for media too. I'm often annoyed with thinks like post `ID` but term `term_id` or other similar things in Polylang related to early decisions.